classes/common.php

```diff
@@ -4,16 +4,29 @@
 defined('MOODLE_INTERNAL') || die();
 
 class common {
-    public static function reset_entry($conditions){
+    public static function reset_entry($conditions, $force = false){
         global $DB;
+
         $oldentry = $DB->get_record('availability_examus', $conditions);
-        if ($oldentry and $oldentry->status != 'Not inited') {
+
+        $not_inited = $oldentry && $oldentry->status == 'Not inited';
+
+        if ($oldentry && (!$not_inited || $force)) {
             $entries = $DB->get_records('availability_examus', [
                 'userid' => $oldentry->userid,
                 'courseid' => $oldentry->courseid,
                 'cmid' => $oldentry->cmid,
-                'status' => 'Not inited']);
-            if (count($entries) == 0) {
+                'status' => 'Not inited'
+            ]);
+
+            if (count($entries) == 0 || $force) {
+                if($force){
+                    foreach($entries as $old){
+                        $old->status = "Force reset";
+                        $DB->update_record('availability_examus', $old);
+                    }
+                }
+
```

Comment:
If count of `$entries` is zero, then `foreach` on `$entries` do nothing. It means, we could safely move this code outside of the `if` clause.

```suggestion
           if($force){
                foreach($entries as $old){
                    $old->status = "Force reset";
                    $DB->update_record('availability_examus', $old);
                }
            }
                
            if (count($entries) == 0 || $force) {


```